src/bika/lims/api/__init__.py

```diff
@@ -1787,3 +1788,21 @@ def is_string(thing):
     :returns: True if the object is a string
     """
     return isinstance(thing, six.string_types)
+
+
+def to_list(value):
+    """Converts the value to a list
+
+    :param value: the value to be represented as a list
+    :returns: a list that represents or contains the value
+    """
+    if is_string(value):
+        try:
+            val = json.loads(value)
+            if isinstance(val, (list, tuple, set)):
+                value = val
+        except (ValueError, TypeError):
+            pass
+    if not isinstance(value, (list, tuple, set)):
+        value = [value]
+    return list(value)
```

Comment:
Please split this function into 2, e.g. something like this:
```python
def parse_json(thing, default=""):
    """Parse from JSON
    """
    try:
        return json.loads(thing)
    except (TypeError, ValueError):
        return default

def to_list(value):
    """Converts the value to a list
    :param value: the value to be represented as a list
    :returns: a list that represents or contains the value
    """
    if is_string(value):
        value = parse_json(value, [])
    if not isinstance(value, (list, tuple, set)):
        value = [value]
    return list(value)
```

Reply:
Done with https://github.com/senaite/senaite.core/pull/2365/commits/4134d9b0a0dd2fdabe58307657460a077fef5a85